packages/next/src/client/components/react-dev-overlay/internal/helpers/hydration-error-info.ts

```diff
@@ -20,9 +20,9 @@ const isHtmlTagsWarning = (msg: NullableText) =>
   Boolean(msg && htmlTagsWarnings.has(msg))
 
 const isTextMismatchWarning = (msg: NullableText) =>
-  Boolean(msg && textMismatchWarnings.has(msg))
+  Boolean(msg && textMismatchWarning === msg)
```

Comment:
Simpler:

```suggestion
  textMismatchWarning === msg
```

Reply:
resolved in [d0c5c59](https://github.com/vercel/next.js/pull/63365/commits/d0c5c595c718faac52cc64342abc150f831fc244)